payroll-processor-functions/src/payroll-processor-functions/Infrastructure/TableQuerier.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using LanguageExt;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace PayrollProcessor.Functions.Infrastructure
+{
+    public class TableQuerier
+    {
+        private readonly CloudTable table;
+
+        public TableQuerier(CloudTable table)
+        {
+            if (table is null)
+            {
+                throw new ArgumentNullException(nameof(table));
+            }
+
+            this.table = table;
+        }
+
+        public async Task<IEnumerable<TModel>> GetAllData<TModel, TEntity>(
+            Func<TEntity, TModel> mapper
+        ) where TEntity : ITableEntity, new()
+        {
+            var query = new TableQuery<TEntity>();
+            var segment = await table.ExecuteQuerySegmentedAsync(query, null);
+
+            return segment.Select(mapper);
+        }
+
+        public async Task<IEnumerable<TModel>> GetAllDataByPartitionKey<TModel, TEntity>(
+            Func<TEntity, TModel> mapper,
+            string partitionKey
+        ) where TEntity : ITableEntity, new()
+        {
+            var filter = TableQuery.GenerateFilterCondition(
```

Comment:
The SDK API uses OData behind the scenes.

I'm still not familiar with how all the querying works or even what the full set of querying options is, but this combination of calls does filter down to a set of entities that all have the same PartitionKey but different RowKeys.